### PR TITLE
Add cons_pal_len

### PR DESCRIPTION
**Pull request by Synray**
Working on #3692

### CODE REVIEW

File: libr/cons/pal.c [hunk 1]
```diff
@@ -534,9 +535,13 @@ R_API RColor r_cons_pal_get_i(int index) {
 
 /* Get color name at index */
 R_API const char *r_cons_pal_get_name(int index) {
-	int i;
-	for (i = 0; i < index && keys[i].name; i++) {}
-	return (i == index) ? keys[index].name : NULL;
+        return (index && index < keys_len) ? keys[index].name : NULL;
```
Comment (radare) [inline libr/cons/pal.c]: use tabs not spaces

File: libr/cons/pal.c [hunk 2]
```diff
@@ -534,9 +535,13 @@ R_API RColor r_cons_pal_get_i(int index) {
...
+}
+
+/* Get number of colors */
+R_API const int r_cons_pal_len()
```
Comment (radare) [inline libr/cons/pal.c]: open brace at right

File: libr/core/disasm.c [hunk 3]
```diff
@@ -328,10 +328,12 @@ static ut64 p2v(RDisasmState *ds, ut64 addr) {
 	return addr;
 }
 
+
```
Comment (radare) [inline libr/core/disasm.c]: unnecessary newline

File: libr/core/disasm.c [hunk 4]
```diff
@@ -328,10 +328,12 @@ static ut64 p2v(RDisasmState *ds, ut64 addr) {
...
 static RAnalFunction *fcnIn(RDisasmState *ds, ut64 at, int type) {
 	if (ds->fcn && r_tinyrange_in (&ds->fcn->bbr, at)) {
 		return ds->fcn;
 	}
+
```
Comment (radare) [inline libr/core/disasm.c]: unnecessary newline

File: libr/core/vmenus.c [hunk 5]
```diff
@@ -3193,7 +3193,7 @@ R_API void r_core_visual_colors(RCore *core) {
 		case 'k': opt--; break;
 		case 'j': opt++; break;
 		case 'K': opt=0; break;
-		case 'J': opt=0; break; // XXX must go to end
+		case 'J': opt=r_cons_pal_len()-1; break;
```
Comment (radare) [inline libr/core/vmenus.c]: add spaces around = and around -

File: libr/core/vmenus.c [hunk 6]
```diff
@@ -3181,8 +3181,8 @@ R_API void r_core_visual_colors(RCore *core) {
 		ch = r_cons_arrow_to_hjkl (ch);
 		switch (ch) {
 #define CASE_RGB(x,X,y) \
-	case x:y--;if(y>0x7f)y=0;break;\
-	case X:y++;if(y>15)y=15;break;
+	case x:if(y>0x00)y--;break;\
```
Comment (radare) [inline libr/core/vmenus.c]: add spaces around > and add {} they are mandatory now

File: libr/cons/pal.c [hunk 7]
```diff
@@ -534,9 +535,13 @@ R_API RColor r_cons_pal_get_i(int index) {
 
 /* Get color name at index */
 R_API const char *r_cons_pal_get_name(int index) {
-	int i;
-	for (i = 0; i < index && keys[i].name; i++) {}
-	return (i == index) ? keys[index].name : NULL;
+        return (index && index < keys_len) ? keys[index].name : NULL;
+}
+
+/* Get number of colors */
```
Comment (radare) [inline libr/cons/pal.c]: obvious comment

File: libr/core/disasm.c (hunk 4 above)
Comment (Synray) [reply libr/core/disasm.c]: Oops my bad

**Pull request merged**